addons/website_slides/controllers/main.py

```diff
@@ -526,8 +526,10 @@ def channel(self, channel=False, channel_id=False, category=None, category_id=Fa
 
         # Check access rights
         if channel_id and not channel:
-            channel = request.env['slide.channel'].browse(channel_id).exists()
```

Comment:
Would make it even more explicit, something like this:
```py
if channel_id < 0:
    # the string part of the channel "slugification" can be blank
    # meaning it can be "/slides/taking-care-of-trees-2" OR just "/slides/-2" if the first part is blank
    # as we use a IntConverter on the route definition, this will pick up a negative ID
    # (the IntConverter is necessary as we want a custom page in case the user can't access the course)
    channel_id = abs(channel_id)
```